starters/cms/src/utils.ts

```diff
@@ -0,0 +1,30 @@
+function slugify(text: string) {
+    return text
+        .replace(/^\s+|\s+$/g, "")
+        .toLowerCase()
+        .replace(/[^a-z0-9 -]/g, "")
+        .replace(/\s+/g, "-")
+        .replace(/-+/g, "-")
+}
+
+export function createUniqueSlug(text: string, existingSlugs: Map<string, number>) {
+    text = text.trim().slice(0, 60)
+
+    if (existingSlugs.has(text)) {
+        const count = existingSlugs.get(text) ?? 0
+        existingSlugs.set(text, count + 1)
+        text = `${text} ${count + 1}`
+    } else {
+        existingSlugs.set(text, 0)
+    }
+
+    return slugify(text)
+}

```

Comment:
This can still result in duplicate slugs, because we check the input instead of the generated slug

Reply:
True, i've made it more robust now. First, input get slugified then function makes it unique.
https://github.com/framer/plugins/pull/222/commits/a3e3b8c8e4b6ff9511f5bda6152ba3a8e0641bc5#diff-e1867fda29cfd75d20d909c3895b73c2878e32c5c28bc70912a893251891c2a1

Reply:
The function still has bugs. And I'm wondering if automatically making it unique is better than telling the user about the duplicate slug so they can fix it themselves. Having a duplicate slug is probably a mistake and if we don't warn them you will publish a URL that you will probably need to change once you see the mistake.

Reply:
The bug is that we don't check the final generated slug for uniqueness. So if my data source has the following slugs it will result in duplicates:
1. `my-slug`
2. `my-slug-0`
3. `my-slug` → transformed into `my-slug-0` which is a duplicate